lib/logger.js

```diff
@@ -1,7 +1,57 @@
 'use babel'
 
+import _ from 'lodash'
+
 export default class Logger {
-  error (/* statusCode, result, builder */) {}
-  warning (/* message */) {}
-  info (/* message */) {}
+  constructor () {
+    this.messages = []
+    this._group = null
+    const callback = () => { this.showFilteredMessages() }
+    atom.config.onDidChange('latex.loggingLevel', callback)
```

Comment:
Good catch!

Initially, the reason that I was listening for it was in cause the user ran a build with the logging level set to "Error" then after the build completed, wanted to see the warnings or info messages already generated without re-running the build. That would make toggling the log level dynamically via a menu command possible.

I am okay with either. If we do not maintain the historical messages beneath the current logging level then I think that code that detects what the current logging level is, would probably need to move into the log parser.